### ISSUE TITLE
result.rb is broken in ruby 2.4.9 "TypeError: {:keyword_init=>true} is not a symbol"

### ISSUE THREAD

**Issue by david-harkness** [alexford/twilito#4]
# result.rb is broken
Ruby 2.4.9

### Short version
```
result = Struct.new(:success?, :errors, :sid, :response, keyword_init: true) {}
=>
TypeError: {:keyword_init=>true} is not a symbol
```

https://stackoverflow.com/questions/15055852/how-to-change-the-default-value-of-a-struct-attribute

### Full File
```
$ irb
2.4.9 :001 > Result = Struct.new(:success?, :errors, :sid, :response, keyword_init: true) do
2.4.9 :002 >         def self.success(**args)
2.4.9 :003?>           new(success?: true, **args)
2.4.9 :004?>         end
2.4.9 :005?>   
2.4.9 :006 >         def self.failure(**args)
2.4.9 :007?>           new(success?: false, **args)
2.4.9 :008?>         end
2.4.9 :009?>   
2.4.9 :010 >         def errors
2.4.9 :011?>           to_h[:errors] || []
2.4.9 :012?>         end
2.4.9 :013?>   
2.4.9 :014 >         def data
2.4.9 :015?>           JSON.parse(response_body || '{}')
2.4.9 :016?>         end
2.4.9 :017?>   
2.4.9 :018 >         private
2.4.9 :019?>   
2.4.9 :020 >         def response_body
2.4.9 :021?>           return nil unless to_h[:response]&.respond_to?(:read_body)
2.4.9 :022?>     
2.4.9 :023 >             to_h[:response].read_body
2.4.9 :024?>         end
2.4.9 :025?>     end
TypeError: {:keyword_init=>true} is not a symbol

```

# Possible fix 
### (from SO link)
```
A = Struct.new(:a, :b, :c) do
  def initialize(a:, b: 2, c: 3)
    super(a, b, c)
  end
end
```

**Comment by alexford** [comment]
Thanks—I've only tested this with Ruby 2.6. I'll try that patch and also see about running tests across Ruby versions.

**Comment by alexford** [comment]
Thanks @david-harkness, this has been fixed in [v0.2.0](https://github.com/alexford/twilito/releases/tag/v0.2.0)
